Core/Epaper/Src/epaper.c

```diff
@@ -0,0 +1,462 @@
+        xSemaphoreTake(epdBusySemHandle, 0); // Guarantee epdBusySemHandle set to 0 
+        if (!(xSemaphoreTake(epdBusySemHandle, 5000) == pdTRUE))
+            {
+                Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+            }
+        Debug("e-Paper busy release\r\n");
+    }
+}
+
+/*!
+  @brief   Send a command to the epaper
+  @param   command Register address of command 
+*/
+static void sendCommand(uint8_t command)
+{
+	// Set Epaper to command mode
+    setCommandMode();
+	// Open spi communication
+    csOn();
+
+	// Write command to spi 
+    HAL_SPI_Transmit_IT(&hspi2, (uint8_t *)&command, 1);
+	// Wait for spi interupt signaling data transfer complete - timeout 10ms
+    if (!(xSemaphoreTake(epdSpiSemHandle, 10) == pdTRUE))
+	{
+		Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+	}
+
+	// Close spi communication
+    csOff();
+}
+
+/*!
+  @brief   Send a command followed by data to the epaper
+  @param   command 	Register address of command 
+  @param   data 	Data to be written to epaper 
+  @param   numBytes Size of data transfer excluding command byte 
+*/
+static void sendMessage(uint8_t command, uint8_t* data, uint16_t numBytes)
+{
+	    // Set Epaper to command mode
+		setCommandMode(); 
+		// Open spi communication	
+		csOn();
+
+		// Write command to spi 
+		HAL_SPI_Transmit_IT(&hspi2, (uint8_t *)&command, 1);
+		// Wait for spi interupt signaling data transfer complete - timeout 10ms
+		if (!(xSemaphoreTake(epdSpiSemHandle, 10) == pdTRUE))
+		{
+			Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+		}
+
+		// Set Epaper to data mode
+		setDataMode();
+
+		// Write command to spi
+		HAL_SPI_Transmit_IT(&hspi2, data, numBytes);
+		// Wait for spi interupt signaling data transfer complete - timout 50ms
+		if (!(xSemaphoreTake(epdSpiSemHandle, 50) == pdTRUE))
+		{
+			Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+		}
+
+		// Close spi communication
+		csOff();
+}
+
+/*!
+  @brief   Set Epaper internal lookup table and voltage data
+  @param   lut 	Look up table to write to epaper
+*/
+static void setLookupTable(uint8_t *lut)
+{
+	// Set look up table register - last 6 bytes belong to seperate registers
+	sendMessage(CMD_WRITE_LUT_REGISTER, lut, LUT_SIZE - 6);
+	// Wait for busy_release interupt
+	waitBusyRelease(); 
+
+	// Set Mystery Register 1
+	sendBuffer[0] = lut[153];		
+	sendMessage(CMD_WRITE_MYSTERY_REGISTER_2, sendBuffer, 1);
+
+	// Set Gate voltage Control Register
+	sendBuffer[0] = lut[154];		
+	sendMessage(CMD_GATE_DRIVING_VOLTAGE_CONTROL, sendBuffer, 1);
+
+	// Set Source Driving Voltage Register
+	sendBuffer[0] = lut[155];
+	sendBuffer[1] = lut[156];
+	sendBuffer[2] = lut[157];
+	sendMessage(CMD_SOURCE_DRIVING_VOLTAGE_CONTROL, sendBuffer, 3);
+
+	// Set VCOM Register
+	sendBuffer[0] = lut[158];
+	sendMessage(CMD_WRITE_VCOM_REGISTER, sendBuffer, 1);
+}
+
+/*!
+  @brief   Turn on Epaper display
+*/
+static void turnOnDisplay()
```

Comment:
Is there potentially a better function name than turnOnDisplay()? Does this capture it's purpose well enough? If this sends the update sequence to the epaper would updateDisplay() be a better name?